backend/src/main.cpp

```diff
@@ -149,74 +150,51 @@ Napi::Boolean AddFrequency(const Napi::CallbackInfo& info)
-    mClient->SetCrossCoupleAcross(frequency, false);
-    mClient->RemoveFrequency(frequency);
-    MainThreadShared::mApiServer->handleAFVEventForWebsocket(
-        sdk::types::Event::kFrequencyStateUpdate, {}, {});
+    RadioState newState;
+
+    newState.frequency = info[0].As<Napi::Number>().Int32Value();
+    newState.rx = false;
+    newState.tx = false;
+    newState.xc = false;
+    newState.headset = false;
+    newState.xca = false;
+
+    RadioHelper::SetRadioState(MainThreadShared::mApiServer.get(), newState);
+    mClient->RemoveFrequency(newState.frequency);
 }
 
 void Reset(const Napi::CallbackInfo& /*info*/) { mClient->reset(); }
 
 Napi::Boolean SetFrequencyState(const Napi::CallbackInfo& info)
 {
-    if (!mClient->IsVoiceConnected()) {
-        return Napi::Boolean::New(info.Env(), false);
-    }
-
-    int frequency = info[0].As<Napi::Number>().Int32Value();
-    if (!mClient->IsFrequencyActive(frequency)) {
-        return Napi::Boolean::New(info.Env(), false);
-    }
-
-    bool rx = info[1].As<Napi::Boolean>().Value();
-    bool tx = info[2].As<Napi::Boolean>().Value();
-    bool xc = info[3].As<Napi::Boolean>().Value();
-    bool onSpeaker = info[4].As<Napi::Boolean>().Value();
-    bool crossCoupleAcrossFrequencies = info[5].As<Napi::Boolean>().Value(); // Not used
-
-    bool oldRxValue = mClient->GetRxState(frequency);
-
-    mClient->SetRx(frequency, rx);
-    mClient->SetRadioGainAll(UserSession::currentRadioGain);
-    if (UserSession::isATC) {
-        mClient->SetTx(frequency, tx);
-        mClient->SetXc(frequency, xc);
-        mClient->SetCrossCoupleAcross(frequency, crossCoupleAcrossFrequencies);
-    } else {
-        mClient->SetTx(frequency, false);
-        mClient->SetXc(frequency, false);
-        mClient->SetCrossCoupleAcross(frequency, false);
-    }
-
-    mClient->SetOnHeadset(frequency, !onSpeaker);
+    RadioState newState;
 
-    MainThreadShared::mApiServer->handleAFVEventForWebsocket(
-        sdk::types::Event::kFrequencyStateUpdate, {}, {});
-
-    if (!oldRxValue && rx) {
-        // When turning on RX, we refresh the transceivers
-        auto states = mClient->getRadioState();
-        if (states.find(frequency) != states.end() && !states[frequency].stationName.empty()) {
-            mClient->FetchTransceiverInfo(states[frequency].stationName);
-        }
-    }
+    newState.frequency = info[0].As<Napi::Number>().Int32Value();
+    newState.rx = info[1].As<Napi::Boolean>().Value();
+    newState.tx = info[2].As<Napi::Boolean>().Value();
+    newState.xc = info[3].As<Napi::Boolean>().Value();
+    // Note the negation here, as the API uses the opposite of what is saved internally
+    newState.headset = !info[4].As<Napi::Boolean>().Value();
+    newState.xca = info[5].As<Napi::Boolean>().Value(); // Not used
 
-    return Napi::Boolean::New(info.Env(), true);
+    auto result = RadioHelper::SetRadioState(MainThreadShared::mApiServer.get(), newState);
```

Comment:
Use the new shared method for setting state.

Reply:
Same here for raw pointers